tests/dataplicity/m2m/test_bencode.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8
+from dataplicity.m2m.bencode import encode, decode, EncodingError, DecodeError
+import pytest
+import six
+
+
+def test_bencode_encoder():
+    """ test bencoding capabilities
+    """
+    assert encode({}) == six.b('de')  # empty dict
+    assert encode({b'foo': 'bar'}) == six.b('d3:foo3:bare')
+    assert encode({
+        b'foo': 'bar',
+        b'fooo': 'bbar'
+    }) == six.b('d3:foo3:bar4:fooo4:bbare')
+    assert encode([]) == six.b('le')  # empty list
+    assert encode(()) == six.b('le')  # empty list
+    assert encode([1, 2, 3]) == six.b('li1ei2ei3ee')
+    assert encode([1, 'foo']) == six.b('li1e3:fooe')
+    assert encode(1) == six.b('i1e')
+    # make sure that the numbers are encoded as bytes. This line would always
+    # return true in python2, but will have a meaning in python3
+    assert isinstance(encode(1), bytes)
+    # utf-8 string
+    assert encode(b'a\xc5\xbc') == six.b('3:a\xc5\xbc')
+    assert encode("aż") == six.b("3:a\xc5\xbc")
+    with pytest.raises(EncodingError):
+        encode(1.38)
+    # since bytes == str in python2, this exception won't be raised
+    if six.PY3:
```

Comment:
Do you need the condition there? Would the following not be appropriate for both?

```
encode({u'foo': u'bar'})
```